jupyterlab_gcedetails/src/service/notebooks_service.ts

```diff
@@ -340,114 +275,102 @@ export class NotebookInstanceServiceLayer {
     return this.metadataPromise;
   }
 
-  /**
-   * Stops the notebook instance.
-   */
-  async stop(): Promise<Instance> {
+  private async submitRequest(
+    request: ApiRequest,
+    errorMessage: string
+  ): Promise<Instance> {
     try {
-      const [projectId, instanceName, locationId] = await Promise.all([
-        this.projectId,
-        this.instanceName,
-        this.locationId,
-      ]);
-      const name = `projects/${projectId}/locations/${locationId}/instances/${instanceName}`;
-      const pendingOperation = await this._transportService.submit<Operation>({
-        path: `${NOTEBOOKS_API_PATH}/${name}:stop`,
-        method: POST,
-        headers: { Authorization: `Bearer ${this._authToken}` },
-        body: {},
-      });
+      const pendingOperation = await this._transportService.submit<Operation>(
+        request
+      );
       const finishedOperation = await this._pollOperation(
         `${NOTEBOOKS_API_PATH}/${pendingOperation.result.name}`
       );
       return finishedOperation.response as Instance;
     } catch (err) {
-      console.error('Unable to stop notebook instance');
+      console.error(errorMessage);
       handleApiError(err);
     }
   }
 
+  /**
+   * Stops the notebook instance.
+   */
+  async stop(): Promise<Instance> {
+    const [projectId, instanceName, locationId] = await Promise.all([
```

Comment:
The block that gets the instance name is repeated in each method. Sorry if my comments in Chat were not clear but this is a candidate for something that I would put into a private method since it is repeatedly used.